### PR TITLE
NXP-30463: Fix Azure storage tests + fix Azure GC + add concurrent image upload test

### PR DESCRIPTION
**Pull request by ataillefer**
The bug described in [NXP-30463](https://jira.nuxeo.com/browse/NXP-30463) only happens in 10.10, with an old version of the `azure-storage` library.
Yet, this PR updates/cleans up/fixes the existing tests and adds a test for concurrent image upload (NXP-30463).

### CODE REVIEW

File: modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-azure/src/test/resources/log4j2-test.xml [hunk 1]
```diff
@@ -12,6 +12,7 @@
   </Appenders>
   <Loggers>
     <Logger name="org.nuxeo" level="info" />
+    <Logger name="org.nuxeo.ecm.blob.azure" level="debug" />
```
Comment (efge) [inline modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-azure/src/test/resources/log4j2-test.xml:15]: Did you really want to commit this? It's bad practice to have verbose logs during tests.

**Review by efge — COMMENTED** [review]
Can the test image be smaller than 3MB?

**Comment by ataillefer** [comment]
> Can the test image be smaller than 3MB?

Not really, a certain size is required for the error to appear with the old lib version, I'd prefer to leave it.

File: modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-azure/src/main/java/org/nuxeo/ecm/blob/azure/AzureGarbageCollector.java [hunk 2]
```diff
@@ -54,26 +51,24 @@ public String getId() {
...
 
-            for (ListBlobItem item : lbs.getResults()) {
-
-                if (!(item instanceof CloudBlockBlob)) {
-                    // ignore subdirectories
-                    continue;
-                }
-
+            // ignore subdirectories by considering only instances of CloudBlockBlob
+            lbs.getResults().stream().filter(item -> (item instanceof CloudBlockBlob)).forEach(item -> {
```
Comment (kevinleturc) [inline modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-azure/src/main/java/org/nuxeo/ecm/blob/azure/AzureGarbageCollector.java:71]:
```suggestion
            lbs.getResults().stream().filter(CloudBlockBlob.class::isInstance).forEach(item -> {
```

File: modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-azure/src/test/resources/log4j2-test.xml (hunk 1 above)
Comment (ataillefer) [reply modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-azure/src/test/resources/log4j2-test.xml:15]: Yes, it's to have the interesting logs from `AzureFileStorage` and they are only appended to the log file, not the console.

**Review by kevinleturc — APPROVED** [review]

**Review by kevinleturc — DISMISSED** [review]

**Review by kevinleturc — APPROVED** [review]

**Review by efge — APPROVED** [review]

**Pull request merged**